File: products/admin-desktop/g5-admin/src/api/client/sites.ts
import type { SiteAddInput } from "../../types/SiteAddInput";
import type { SiteCatalog } from "../../types/SiteCatalog";
import type { SiteDeleteInput } from "../../types/SiteDeleteInput";
import type { SiteHealthCheckInput } from "../../types/SiteHealthCheckInput";
import type { SiteHealthCheckResult } from "../../types/SiteHealthCheckResult";
import type { SiteSwitchInput } from "../../types/SiteSwitchInput";
import type { SiteUpdateInput } from "../../types/SiteUpdateInput";
import type { SiteActivityListResponse } from "../../types/SiteActivityListResponse";
import { invokeCommand } from "./core";

export async function getSiteCatalog(): Promise<SiteCatalog> {
  return invokeCommand<SiteCatalog>("cmd_site_catalog_get");
}

export async function addSite(input: SiteAddInput): Promise<SiteCatalog> {
  return invokeCommand<SiteCatalog>("cmd_site_add", { input });
}

export async function updateSite(input: SiteUpdateInput): Promise<SiteCatalog> {
  return invokeCommand<SiteCatalog>("cmd_site_update", { input });
}

export async function deleteSite(input: SiteDeleteInput): Promise<SiteCatalog> {
  return invokeCommand<SiteCatalog>("cmd_site_delete", { input });
}

export async function switchSite(input: SiteSwitchInput): Promise<SiteCatalog> {
  return invokeCommand<SiteCatalog>("cmd_site_switch", { input });
}

export async function healthCheckSite(
  input: SiteHealthCheckInput,
): Promise<SiteHealthCheckResult> {
  return invokeCommand<SiteHealthCheckResult>("cmd_site_health_check", { input });
}

export async function getSiteActivityList(
  siteId?: string | null,
  limit = 12,
): Promise<SiteActivityListResponse> {
  return invokeCommand<SiteActivityListResponse>("cmd_site_activity_list", {
    limit,
    site_id: siteId ?? null,
  });
}

File: g5-admin/src-tauri/src/app_state/sites.rs
#[cfg(test)]
use super::*;

#[cfg(test)]
#[allow(dead_code)]
impl AppState {
    pub async fn site_catalog(&self, request_id: &str) -> Result<SiteCatalog, AppError> {
        self.ensure_master_unlocked().await?;
        self.site_catalog_service().catalog(request_id).await
    }

    pub async fn active_site(&self) -> Result<Option<Site>, AppError> {
        self.site_catalog_service().active_site().await
    }

    pub async fn switch_site(&self, site_id: &str) -> Result<(), AppError> {
        self.ensure_master_unlocked().await?;
        self.site_catalog_service().switch_site(site_id).await
    }

    pub async fn add_site(&self, input: SiteAddInput) -> Result<Site, AppError> {
        self.ensure_master_unlocked().await?;
        self.site_catalog_service().add_site(input).await
    }

    pub async fn update_site(&self, input: SiteUpdateInput) -> Result<Site, AppError> {
        self.ensure_master_unlocked().await?;
        self.site_catalog_service().update_site(input).await
    }

    pub async fn delete_site(&self, input: SiteDeleteInput) -> Result<(), AppError> {
        self.site_catalog_service().delete_site(&input).await
    }

    pub async fn activity_list(
        &self,
        request_id: &str,
        site_id: Option<String>,
        limit: usize,
    ) -> Result<SiteActivityListResponse, AppError> {
        self.site_catalog_service()
            .activity_list(request_id, site_id, limit)
            .await
    }
}
